app/serializers/api/user/subscription_serializer.rb

```diff
@@ -12,6 +12,18 @@ class SubscriptionSerializer
       attribute :subscription_type do |subscription|
         subscription.subscription_type.name
       end
+
+      attribute :meta do |subscription|
+        service_class = "Api::User::#{subscription.subscription_type.name.camelize}MetaService".safe_constantize
+        if service_class
+          service_class.new(
+            subscription.metadata['commodity_codes'],
+            subscription.subscription_targets_dataset.commodities.map(&:target_id),
+          ).call
```

Comment:
I think you can just pass the subscription object to this service and let it use what it needs.